frontend/src/card/card.controllers.js

```diff
@@ -891,19 +898,9 @@ CardControllers.controller('CardDetail', [
             }
         });
 
-        // TODO: while we wait for the databases list we should put something on screen
-        // grab our database list, then handle the rest
-        async function loadDatabasesAndTables() {
-            let dbs = await Metabase.db_list().$promise;
-            return await * dbs.map(async function(db) {
-                db.tables = await Metabase.db_tables({ dbId: db.id }).$promise;
-                return db;
-            });
-        }
-
         async function init() {
             try {
-                databases = await loadDatabasesAndTables();
+                databases = await Metabase.db_list_with_tables().$promise;
```

Comment:
Can we just grab all the database metadata here and cut out quite a few more api calls?  I think it would be more ideal to move to a model where the entire set of metadata for a database is returned in a single call and then we don't need to worry about any extra api calls at all.  plus users don't have to do any waiting as they make choices in the UI.